src/ImageSharp/Common/Helpers/Guard.cs

```diff
@@ -17,13 +17,14 @@ internal static partial class Guard
     /// <typeparam name="TValue">The type of the value.</typeparam>
     /// <exception cref="ArgumentException"><paramref name="value"/> is not a value type.</exception>
     [MethodImpl(InliningOptions.ShortMethod)]
-    public static void MustBeValueType<TValue>(TValue value, string parameterName)
+    public static void SamplerMustBeValueType<TValue>(TValue value, [CallerArgumentExpression("value")] string? parameterName = null)
```

Comment:
I went in Renaming this method because it was just used for samplers to make sure that the implementation is a struct and not a class.

So I added IResampler as a Type constraint.

Reply:
Guard types are generally meant to be more general in nature. It's only used by the sampler now but could be used for other things in the future.

Reply:
I found a good solution ^^